pkg/didcomm/protocol/didexchange/persistence/connection_test.go

```diff
@@ -36,61 +38,84 @@ func Test_ComputeHash(t *testing.T) {
 }
 
 func TestConnectionRecord_SaveInvitation(t *testing.T) {
-	store := &storage.MockStore{Store: make(map[string][]byte)}
-	record := NewConnectionRecorder(store)
-	require.NotNil(t, record)
-
-	key := "sample-key"
-	value := struct {
-		Code    int
-		Message string
-	}{
-		Code:    1,
-		Message: "sample-msg",
-	}
-
-	err := record.SaveInvitation(key, value)
-	require.NoError(t, err)
-
-	require.NotEmpty(t, store)
-
-	k, err := computeHash([]byte(key))
-	require.NoError(t, err)
-	require.NotEmpty(t, k)
-
-	v, err := record.store.Get(k)
-	require.NoError(t, err)
-	require.NotEmpty(t, v)
+	t.Run("test save invitation success", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+
+		key := "sample-key1"
+		value := &models.Invitation{
+			ID:    "sample-id1",
+			Label: "sample-label1",
+		}
+
+		err := record.SaveInvitation(key, value)
+		require.NoError(t, err)
+
+		require.NotEmpty(t, store)
+
+		k, err := invitationKey(key)
+		require.NoError(t, err)
+		require.NotEmpty(t, k)
+
+		v, err := record.store.Get(k)
+		require.NoError(t, err)
+		require.NotEmpty(t, v)
+	})
+
+	t.Run("test save invitation failure due to invalid key", func(t *testing.T) {
+		store := &mockstorage.MockStore{Store: make(map[string][]byte)}
+		record := NewConnectionRecorder(store)
+		require.NotNil(t, record)
+
+		value := &models.Invitation{
+			ID:    "sample-id2",
+			Label: "sample-label2",
+		}
+		err := record.SaveInvitation("", value)
+		require.Error(t, err)
```

Comment:
It is better to check for exact error, not just error

Reply:
done